### PR TITLE
Update fullstack challenge for FY24

### PR DESCRIPTION
**Pull request by zmb3**
This change aims to simplify the fullstack challenge by addressing some common areas of confusion and reducing scope.

Since we're currently only seeking L4 candidates, the requirements for other levels have been temporarily removed. We will make similar improvements when we have a need for more fullstack engineers at other levels.

Also clean up a few other details while we're here (countries that we hire in, terminology, etc)

### CODE REVIEW

File: challenges/fullstack/challenge.md [hunk 1]
```diff
@@ -19,148 +14,124 @@ We would like to evaluate your skill in the following areas:
...
-This level focuses on serving web assets. In addition to the Level 1
-requirements, you will:
+The starter code here only listens for plain-text HTTP. At Teleport, we avoid
+plain-text connections and prefer to encrypt all data in transit. You will be
+responsible for adding TLS.
 
-* Serve web assets from a web server.
-* Containerize your application using Docker.
-* Implement URL Navigation
+Your API only needs to return information about a single directory. There is no
```
Comment (russjones) [inline challenges/fullstack/challenge.md:75]: Right now it sounds like the API always returns one static directory. Might want to make it clear that the user can request a directory listing for any directory, but the content listing for that directory only needs to be contents of that directory and not recursive listing.

To be honest, typing this out in a clear way was hard for me, so maybe an example would help here?

File: challenges/fullstack/challenge.md [hunk 2]
```diff
@@ -19,148 +14,124 @@ We would like to evaluate your skill in the following areas:
...
-fullstack challenge.
+There are 6 engineering levels at Teleport. Levels 1-4 are open to external candidates.
 
 For fullstack engineers, levels 5 and 6 are only for internal promotions. Check
 [Fullstack Engineering Levels](../../levels/fullstack.pdf) for more details.
 
-## Requirements
+At this time, we are seekling level 4 full stack engineers only.
+
+## Requirements - Level 4
```
Comment (russjones) [inline challenges/fullstack/challenge.md:54]: Should we call out the security requirements in this block? I know they're mentioned later, but mitigating against common web security issues feels like it should be highlighted.

File: challenges/fullstack/challenge.md [hunk 3]
```diff
@@ -19,148 +14,124 @@ We would like to evaluate your skill in the following areas:
...
+  and URL-based navigation.
+* Strong authentication
 
-<img src="./assets/google.jpg" height="300" />
-<img src="./assets/github.jpg" height="200" />
+### API
 
-### Level 2
+This repository includes just enough starter code to serve up both the web app
+and a sample API endpoint.
```
Comment (russjones) [inline challenges/fullstack/challenge.md:69]: Where in the careers repo is the starter code?

**Review by russjones — APPROVED** [review]
Overall looks good to me, some minor comments.

File: challenges/fullstack/challenge.md (hunk 3 above)
Comment (zmb3) [reply challenges/fullstack/challenge.md:69]: I'll clean this up. It's a separate repo.

**Pull request merged**